### ISSUE TITLE
Error: Multiple exported component definitions found

### ISSUE THREAD

**Issue by stuff** [storybooks/storybook#1898]
When exporting a component as a named export and default exporting the same component using an HOC in the same file, I've got a bunch of errors like this the **first time** I run `start-storybook`:
```
=> Using default webpack setup based on "Create React App".
 16% building modules 57/73 modules 16 active ...el-runtime/core-js/symbol/iterator.jsError: Multiple exported component definitions found.
    at Context.exportDeclaration (/Users/nicolas.challeil/test-storybook/node_modules/react-docgen/dist/resolver/findExportedComponentDefinition.js:116:13)
    at Context.invokeVisitorMethod (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path-visitor.js:344:49)
    at Visitor.PVp.visitWithoutReset (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path-visitor.js:196:32)
    at NodePath.each (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path.js:101:26)
    at visitChildren (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path-visitor.js:219:18)
    at Visitor.PVp.visitWithoutReset (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path-visitor.js:204:20)
    at visitChildren (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path-visitor.js:246:25)
    at Visitor.PVp.visitWithoutReset (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path-visitor.js:204:20)
    at Visitor.PVp.visit (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path-visitor.js:133:29)
    at Object.visit (/Users/nicolas.challeil/test-storybook/node_modules/ast-types/lib/path-visitor.js:101:55)
```
But the storybook **is created and is working** (however, in our big project, with a lot of snapshots, the deploy machine crashes because of too many errors like this apparently)

Second time, a `.cache` folder has appeared in `node_modules` and no errors are raised.

Removing the folder `.cache` and re starting storybook will log the errors again.

i've created a github repo test here `https://github.com/stuff/test-storybook`

here is my component:
```js
import React, { Component } from 'react';

const hoc = (WrappedComponent) => {
  class HOC extends Component {
    render() {
      return (
        <WrappedComponent />
      );
    }
  }

  return HOC;
}

export class MyComponent1 extends Component {
  render() {
    return (
      <div>Test</div>
    );
  }
}

export default hoc(MyComponent1);
```

**Steps to reproduce**
* git clone https://github.com/stuff/test-storybook
* cd test-storybook
* yarn install
* yarn run storybook

**Comment by danielduan** [comment]
That's an error from `react-docgen` and probably related to `babel-plugin-react-docgen` passes things in.

I think you'll need to use a custom babel file and export all component definitions: https://github.com/storybooks/babel-plugin-react-docgen#babelrc-options

**Comment by vladi-dev** [comment]
Versions:
```
"@storybook/addon-actions": "^3.2.11",
"@storybook/addon-info": "^3.2.11",
"@storybook/addon-links": "^3.2.10",
"@storybook/react": "^3.2.11",
```
After updating to v3.2.11 I receive multiple similar errors when running `npm run storybook`.
```
No suitable component definition found.
    at parse (/Users/username/remote-client/node_modules/react-docgen/dist/parse.js:78:9)
    at Object.defaultParse [as parse] (/Users/username/remote-client/node_modules/react-docgen/dist/main.js:66:30)
    at injectReactDocgenInfo (/Users/username/remote-client/node_modules/babel-plugin-react-docgen/lib/index.js:166:24)
    at PluginPass.FunctionDeclarationFunctionExpressionArrowFunctionExpression (/Users/username/remote-client/node_modules/babel-plugin-react-docgen/lib/index.js:76:9)
    at newFn (/Users/username/remote-client/node_modules/babel-traverse/lib/visitors.js:276:21)
    at NodePath._call (/Users/username/remote-client/node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (/Users/username/remote-client/node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (/Users/username/remote-client/node_modules/babel-traverse/lib/path/context.js:105:12)
    at TraversalContext.visitQueue (/Users/username/remote-client/node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitSingle (/Users/username/remote-client/node_modules/babel-traverse/lib/context.js:108:19)
    at TraversalContext.visit (/Users/username/remote-client/node_modules/babel-traverse/lib/context.js:192:19)
```
Downgrading to versions below resolved the issue.
```
"@storybook/addon-actions": "^3.2.6",
"@storybook/addon-info": "^3.2.9",
"@storybook/addon-links": "^3.2.6",
"@storybook/react": "^3.2.8",
```

**Comment by danielduan** [comment]
This error is related to https://github.com/storybooks/babel-plugin-react-docgen/issues/36 and fixed in v1.8.1.

Please delete your node_modules and all lock files and reinstall. Sorry about this guys.
